neo/Storage/Implementation/test/test_db_factory.py

```
@@ -0,0 +1,124 @@
+            same.
+        '''
+        def make_compatible(obj, to):
+            if not isinstance(obj, to):
+                new_obj = to(obj)
+                if isinstance(new_obj, dict):
+                    return new_obj.items()
+                return new_obj
+            return obj
+
+        with self._db.openIter(DBProperties(prefix=b'00001', include_value=True, include_key=False)) as iterator:
+
+            iterator = make_compatible(iterator, list)
+            self.assertEqual(len(iterator), 3)
+            self.assertIsInstance(iterator, list)
+
+        with self._db.openIter(DBProperties(prefix=b'00002', include_value=False, include_key=True)) as iterator:
+            iterator = make_compatible(iterator, list)
+            self.assertEqual(len(iterator), 4)
+            self.assertIsInstance(iterator, list)
+
+        with self._db.openIter(DBProperties(prefix=b'00002', include_value=True, include_key=True)) as iterator:
+            iterator = make_compatible(iterator, dict)
+            self.assertEqual(len(iterator), 4)
+            self.assertIsInstance(iterator, abc.ItemsView)
+
+        with self._db.openIter(DBProperties(prefix=None, include_value=True, include_key=True)) as iterator:
+            iterator = make_compatible(iterator, dict)
+            self.assertEqual(len(iterator), 7)
+            self.assertIsInstance(iterator, abc.ItemsView)
+
+        with self._db.openIter(DBProperties(prefix=None, include_value=False, include_key=True)) as iterator:
+            iterator = make_compatible(iterator, list)
+            self.assertEqual(len(iterator), 7)
+            self.assertIsInstance(iterator, list)
+
+        with self._db.openIter(DBProperties(prefix=None, include_value=True, include_key=False)) as iterator:
+            iterator = make_compatible(iterator, list)
+            self.assertEqual(len(iterator), 7)
+            self.assertIsInstance(iterator, list)
+
```

what happens if we do  `DBProperties` with `include_value=False` and `include_key=False`? I know it doesn't make sense, but we should probably raise an exception as an invalid configuration. This can then be tested here as well

---

Catching it now in the init and there is a test for it.